xdem/spstats.py

```diff
@@ -1003,22 +1009,273 @@ def patches_method(dh : np.ndarray, mask: np.ndarray[bool], gsd : float, area_si
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        ax1.set_xlim((0,1.1*np.max(list_counts)))
+        ax1.set_ylim((np.min(interval_var_2.left),np.max(interval_var_2.right)))
+    ax1.ticklabel_format(axis='x',style='sci',scilimits=(0,0))
+    ax1.spines['top'].set_visible(False)
+    ax1.spines['right'].set_visible(False)
+    # Try to identify if the count is always the same
+    if np.sum(~(np.abs(list_counts[0] - np.array(list_counts)) < 5)) <= 2:
+        ax1.text(0.5, 0.5, "Fixed number of\nsamples: " + '{:,}'.format(int(list_counts[0])), ha='center', va='center',
+                 fontweight='bold', transform=ax1.transAxes, rotation=90, bbox=dict(facecolor='white', alpha=0.8))
+
+    # Third, an axis to plot the data as a colored grid
+    ax = fig.add_subplot(grid[3:, :-3])
+
+    # Define limits of colormap is none are provided, robust max and min using percentiles
+    if vmin is None and vmax is None:
+        vmax = np.nanpercentile(df_sub[statistic_name].values, 99)
+        vmin = np.nanpercentile(df_sub[statistic_name].values, 1)
+
+    # Create custom colormap
+    col_bounds = np.array([vmin, np.mean([vmin,vmax]), vmax])
+    cb = []
+    cb_val = np.linspace(0, 1, len(col_bounds))
+    for j in range(len(cb_val)):
+        cb.append(cmap(cb_val[j]))
+    cmap_cus = colors.LinearSegmentedColormap.from_list('my_cb', list(
+        zip((col_bounds - min(col_bounds)) / (max(col_bounds - min(col_bounds))), cb)), N=1000)
+
+    # Plot a 2D colored grid using fill_between
+    for i in range(len(unique_var_1)):
+        for j in range(len(unique_var_2)):
+            df_both = df_sub[np.logical_and(df_sub.var1_mid == unique_var_1[i], df_sub.var2_mid == unique_var_2[j])]
+
+            stat = df_both[statistic_name].values[0]
+            if np.isfinite(stat):
+                stat_col = max(0.0001,min(0.9999,(stat - min(col_bounds))/(max(col_bounds)-min(col_bounds))))
+                col = cmap_cus(stat_col)
+            else:
+                col = nodata_color
+
+            ax.fill_between([df_both[var_name_1].values[0].left, df_both[var_name_1].values[0].right], [df_both[var_name_2].values[0].left] * 2,
+                            [df_both[var_name_2].values[0].right] * 2, facecolor=col, alpha=1, edgecolor='white')
+
+    ax.set_xlabel(label_var_name_1)
+    ax.set_ylabel(label_var_name_2)
+    ax.set_xscale(scale_var_1)
+    ax.set_yscale(scale_var_2)
+    # In case the axis value does not agree with the scale (e.g., 0 for log scale)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        ax.set_xlim((np.min(interval_var_1.left),np.max(interval_var_1.right)))
+        ax.set_ylim((np.min(interval_var_2.left),np.max(interval_var_2.right)))
+
+    # Fourth and finally, add a colormap and nodata color to the legend
+    axcmap = fig.add_subplot(grid[:3, -3:])
+
+    # Remove ticks, labels, frame
```

Comment:
This kind of function that can be re-used elsewhere could go in a dedicated module for plotting (same as for the colormaps)

Reply:
Yes, I guess this module will be created with all the plots we need to do for the `Coreg` functions!

Reply:
Raised in #190